app/models.py

```diff
@@ -429,14 +429,33 @@ def serialize(self, data=None):
             "frameworkSlug": self.framework.slug,
             "declaration": self.declaration,
             "onFramework": self.on_framework,
-            "agreementReturned": bool(agreement_returned_at),
-            "agreementReturnedAt": agreement_returned_at,
-            "agreementDetails": self.agreement_details,
-            "countersigned": bool(countersigned_at),
-            "countersignedAt": countersigned_at,
             "agreedVariations": agreed_variations,
         }, **(data or {}))
 
+        if self.framework_agreements:
```

Comment:
Where does `self.framework_agreements` come from? Does it match the `supplier_id` and `framework_id` in the framework_agreements table through SQLAlchemy and the magic of foreign keys?